internal/auth0/connection/resource_client.go

```diff
@@ -52,12 +52,12 @@ func NewClientResource() *schema.Resource {
 
 func createConnectionClient(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	api := meta.(*config.Config).GetAPI()
-	mutex := meta.(*config.Config).GetMutex()
 
 	connectionID := data.Get("connection_id").(string)
 
-	mutex.Lock(connectionID)
-	defer mutex.Unlock(connectionID)
+	mutex := meta.(*config.Config).GetMutex()
+	mutex.Lock(connectionID + "-clients") // Prevents colliding API requests between other `auth0_connection_client` resource.
```

Comment:
Appending the `-clients` string to the key makes it unique which makes it clear what other locks its competing with and also makes it unique so that other facets of the connection can be locked on.